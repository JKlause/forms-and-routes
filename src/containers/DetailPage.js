import React, { Component } from 'react';
import DetailView from '../components/DetailView';
import { callApi } from '../service/callApi';
import PropTypes from 'prop-types';

export default class DetailPage extends Component {


  static propTypes = {
    history: PropTypes.object.isRequired,
    match: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.string
      })
    })
  }

  state = {
    character: {},
    loading: false
  }

 
  componentDidMount() {
    this.setState({ loading: true });
    callApi(this.props.match.params.id)
      .then((character) => {
        this.setState({ character, loading: false });
      });
  }


  render() {
    if(this.state.loading) return <img src='https://media.giphy.com/media/w461tXo03sQP6/giphy.gif'></img>;

    return (
      <>
        <DetailView {...this.state.character} />
      </>
    );
  }
}
