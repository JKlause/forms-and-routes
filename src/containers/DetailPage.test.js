import React from 'react';
import { shallow } from 'enzyme';
import DetailPage from './DetailPage';



describe('DetailPage container', () => {

  it('renders DetailPage component', () => {

    // eslint-disable-next-line no-undef
    global.fetch = jest.fn(() => Promise.resolve());
    const wrapper = shallow(<DetailPage match={{ params: { id: '123' } }} history={{}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
