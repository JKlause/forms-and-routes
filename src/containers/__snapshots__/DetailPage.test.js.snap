// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailPage container renders DetailPage component 1`] = `
<img
  src="https://media.giphy.com/media/w461tXo03sQP6/giphy.gif"
/>
`;
